core/pbcc_subscribe_event_engine.c

```diff
@@ -538,6 +554,19 @@ enum pubnub_res pbcc_subscribe_ee_reconnect(
     PUBNUB_ASSERT_OPT(NULL != ee);
 
     pubnub_mutex_lock(ee->mutw);
+    pbcc_ee_state_t* current_state_obj = pbcc_ee_current_state(ee->ee);
+    if (NULL != current_state_obj) {
+        pbcc_subscribe_ee_state current_state_type = (pbcc_subscribe_ee_state)pbcc_ee_state_type(current_state_obj);
+        if (current_state_type == SUBSCRIBE_EE_STATE_RECEIVING ||
+            current_state_type == SUBSCRIBE_EE_STATE_UNSUBSCRIBED) {
+            pbcc_ee_state_free(&current_state_obj);
+            pubnub_mutex_unlock(ee->mutw);
+            PUBNUB_LOG_WARNING("pbcc_subscribe_ee_reconnect: called while in state. Skipping reconnect.\n");
+            return PNR_OK;
+        }
+        pbcc_ee_state_free(&current_state_obj);
+    }
+
```

Comment:
We also can't reconnect while in `handshaking` state.  

> called while in state.  

Should be there some reason explained? 

```suggestion
    pbcc_ee_state_t* current_state_obj = pbcc_ee_current_state(ee->ee);
    pbcc_ee_state_free(&current_state_obj);
    pubnub_mutex_unlock(ee->mutw);

    if (NULL != current_state_obj) {
        pbcc_subscribe_ee_state current_state_type = (pbcc_subscribe_ee_state)pbcc_ee_state_type(current_state_obj);
        if (current_state_type == SUBSCRIBE_EE_STATE_UNSUBSCRIBED ||
            current_state_type == SUBSCRIBE_EE_STATE_HANDSHAKING ||
            current_state_type == SUBSCRIBE_EE_STATE_RECEIVING) {
            PUBNUB_LOG_WARNING("pbcc_subscribe_ee_reconnect: called while in state. Skipping reconnect.\n");
            return PNR_OK;
        }
    }

```

Reply:
Yeah, definitely. I will add those, thank you!